integration-tests/polling-subscribe-provider.spec.ts

```diff
@@ -1,78 +1,322 @@
-        tag: 'first',
-        address: eventContractAddress
+        tag: 'intFromMainContract',
+        address: mainContractAddress,
       });
 
       eventSub.on('data', (x) => {
         data.push(x);
       });
 
-      const contract = await Tezos.contract.at(eventContractAddress!);
-      const op = await contract.methods.default().send();
+      const contract1 = await Tezos.contract.at(mainContractAddress!);
+      const contract2 = await secondUser.contract.at(mainContractAddress!);
+      
+
+      const operation1 = contract1.methodsObject.default({
+        ad: calledContractAddress,
+        newStore: 0,
+        p: 0,
+        t: 0
+      });
+
+      const operation2 = contract2.methodsObject.default({
+        ad: calledContractAddress,
+        newStore: 0,
+        p: 0,
+        t: 0
+      });
 
-      await op.confirmation();
+      const sent1 = await operation1.send();
+      const sent2 = await operation2.send();
+
+      await sent1.confirmation();
+      await sent2.confirmation();
 
       await sleep(3000);
 
       eventSub.close();
 
-      expect(data.length).toEqual(1);
+      expect(data.length).toEqual(2);
       expect(data[0].opHash).toBeDefined();
       expect(data[0].blockHash).toBeDefined();
       expect(data[0].level).toBeDefined();
       expect(data[0].kind).toEqual('event');
-      expect(data[0].source).toEqual(eventContractAddress);
-      expect(data[0].tag).toEqual('first');
+      expect(data[0].source).toEqual(mainContractAddress);
+      expect(data[0].tag).toEqual('intFromMainContract');
       expect(data[0].type).toBeDefined();
       expect(data[0].payload).toBeDefined();
       expect(data[0].result).toBeDefined();
 
       done();
     });
-  })
-})
+
+    it('should include events from failed operations when not filtering', async (done) => {
+      const data: any = [];
+
+      const eventSub = Tezos.stream.subscribeEvent({
+        address: mainContractAddress,
+      });
+
+      eventSub.on('error', (event) => {
+        data.push({
+          type: 'error',
+          event,
+        });
+      });
+
+      eventSub.on('data', (event) => {
+        data.push({
+          type: 'data',
+          event,
+        });
+      });
+
+      const contract1 = await Tezos.contract.at(mainContractAddress!);
+      const contract2 = await secondUser.contract.at(mainContractAddress!);
+      try {
+        const operation1 = contract1.methodsObject.default({
+          ad: calledContractAddress,
+          newStore: 1,
+          p: 0,
+          t: 0
+        });
+  
+        const operation2 = contract2.methodsObject.default({
+          ad: calledContractAddress,
+          newStore: 0,
+          p: 0,
+          t: 0
+        });
+  
+        const sent1 = await operation1.send();
+        const sent2 = await operation2.send();
+  
+        await sent1.confirmation();
+        await sent2.confirmation();
+      } catch (e) {
+        // Failure is expected
```

Comment:
I did that when testing, but because failure is expected, this will pollute the integration tests with output that's not useful to the user (a failure that is expected)